src/components/checkbox/__tests__/index.spec.tsx

```diff
@@ -24,57 +27,253 @@ describe('Checkbox renderer test', () => {
+        checkboxRef.current?.validate();
+        await driver.press();
+
+        expect(onChangeValidity).toHaveBeenCalledWith(true);
+      });
+
+      it('should been called after invoking validate() and value changed to true and then false', async () => {
+        const props = {required: true, onChangeValidity};
+        const component = <TestCase {...props}/>;
+        const driver = new CheckboxDriver({component, testID});
+
+        checkboxRef.current?.validate();
+        await driver.press();
+        await driver.press();
+
+        expect(onChangeValidity).toHaveBeenCalledWith(false);
+      });
+    });
+
+    describe('isValid', () => {
+      it('should be valid if initial value is false', async () => {
+        const props = {};
+        const component = <TestCase {...props}/>;
+        //eslint-disable-next-line
+        const driver = new CheckboxDriver({component, testID});
+
+        const isValid = checkboxRef.current?.isValid();
+
+        expect(isValid).toBe(true);
+      });
+
+      it('should be valid if initial value is true', async () => {
+        const props = {value: true};
+        const component = <TestCase {...props}/>;
+        //eslint-disable-next-line
+        const driver = new CheckboxDriver({component, testID});
+
+        const isValid = checkboxRef.current?.isValid();
+
+        expect(isValid).toBe(true);
+      });
+
+      it('should be valid if initial value is true and is required', async () => {
+        const props = {value: true, required: true};
+        const component = <TestCase {...props}/>;
+        //eslint-disable-next-line
+        const driver = new CheckboxDriver({component, testID});
+
+        const isValid = checkboxRef.current?.isValid();
+
+        expect(isValid).toBe(true);
+      });
+
+      it('should be invalid if initial value is false and is required', async () => {
+        const props = {required: true};
+        const component = <TestCase {...props}/>;
+        //eslint-disable-next-line
+        const driver = new CheckboxDriver({component, testID});
+
+        const isValid = checkboxRef.current?.isValid();
+
+        expect(isValid).toBe(false);
+      });
+
+      it('should be invalid after validate when initial value is false', async () => {
+        const props = {required: true};
+        const component = <TestCase {...props}/>;
+        //eslint-disable-next-line
+        const driver = new CheckboxDriver({component, testID});
+
+        let isValid = checkboxRef.current?.isValid();
+        expect(isValid).toBe(false);
+
+        checkboxRef.current?.validate();
+        isValid = checkboxRef.current?.isValid();
+
+        expect(isValid).toBe(false);
+      });
+
+      it('should be valid after validate when initial value is true', async () => {
+        const props = {value: true, required: true};
+        const component = <TestCase {...props}/>;
+        //eslint-disable-next-line
+        const driver = new CheckboxDriver({component, testID});
```

Comment:
Can we removing these ignores?

Reply:
It doesn't let me remove the const -> Error: Do not use 'new' for side effects.eslint[no-new](https://eslint.org/docs/rules/no-new)

Reply:
Maybe ignore that one, it is more readable IMO